### PR TITLE
Feature/civ 1476 do you have loans or credit card debts

### PR DESCRIPTION
**Pull request by LeonardoPalmeiro**
### JIRA link ###

civ_1476

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/unit/routes/features/response/statementOfMeans/debts/debtsRouter.test.ts [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+  const citizenRoleToken: string = config.get('citizenRoleToken');
+  const idamUrl: string = config.get('idamUrl');
+  beforeEach(() => {
+    nock(idamUrl)
+      .post('/o/token')
+      .reply(200, {id_token: citizenRoleToken});
+  });
+  describe('on Exception', () => {
+    test('should return http 500 when has error in the get method', async () => {
+      const redisFailureError = 'Redis DraftStore failure.';
```
Comment (laarmada) [inline src/test/unit/routes/features/response/statementOfMeans/debts/debtsRouter.test.ts:42]: use constantErrorMessages

File: src/test/unit/routes/features/response/statementOfMeans/debts/debtsRouter.test.ts [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+import {StatementOfMeans} from '../../../../../../../main/common/models/statementOfMeans';
+import {
+  ENTER_A_DEBT,
+  ENTER_AT_LEAST_ONE_DEBT, VALID_NUMBER_OF_PEOPLE, VALID_TWO_DECIMAL_NUMBER,
+  VALID_YES_NO_OPTION,
+} from '../../../../../../../main/common/form/validationErrors/errorMessageConstants';
+
+jest.mock('../../../../../../../main/modules/oidc');
+jest.mock('../../../../../../../main/modules/draft-store');
+jest.mock('../../../../../../../main/modules/draft-store/draftStoreService');
```
Comment (laarmada) [inline src/test/unit/routes/features/response/statementOfMeans/debts/debtsRouter.test.ts:28]: Why mock draftstore service if you have mocked draftstore? It overrides the above mock.....

File: src/main/app.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@ import { OidcMiddleware } from './modules/oidc';
 import {DraftStoreClient} from './modules/draft-store';
 import routes from './routes/routes';
 
+
```
Comment (TatsyMaslova) [inline src/main/app.ts:18]: extra space? lol

**Review by RosDiaf — APPROVED** [review]

File: src/main/common/form/validators/formValidator.ts [hunk 4]
```diff
@@ -2,9 +2,10 @@ import {Validator} from 'class-validator';
 import {Form} from '../models/form';
 
 const validator = new Validator();
-
+const ERROR_FIELD = 'errors';
 export async function validateForm(form: Form) {
   const errors = await validator.validate(form);
+  Object.keys(form);
```
Comment (TatsyMaslova) [inline src/main/common/form/validators/formValidator.ts:8]: where is it used?

File: src/main/routes/features/response/statementOfMeans/debts/debtsRouter.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+import {DebtItems} from '../../../../../common/form/models/statementOfMeans/debts/debtItems';
+import {validateFormNested} from '../../../../../common/form/validators/formValidator';
+import {Claim} from '../../../../../common/models/claim';
+import {getCaseDataFromStore, saveDraftClaim} from '../../../../../modules/draft-store/draftStoreService';
+import {StatementOfMeans} from '../../../../../common/models/statementOfMeans';
+import {YesNo} from '../../../../../common/form/models/yesNo';
+import {constructResponseUrlWithIdParams} from '../../../../../common/utils/urlFormatter';
+
+const debtsViewPath = 'features/response/statementOfMeans/debts/debts';
+const debtsRouter = express.Router();
```
Comment (TatsyMaslova) [inline src/main/routes/features/response/statementOfMeans/debts/debtsRouter.ts:13]: did we agree to call this controller?

File: src/main/app.ts (hunk 3 above)
Comment (LeonardoPalmeiro) [reply src/main/app.ts:18]: thanskkk
done.

File: src/main/common/form/validators/formValidator.ts (hunk 4 above)
Comment (LeonardoPalmeiro) [reply src/main/common/form/validators/formValidator.ts:8]: line 26       if (key !== ERROR_FIELD && form[key] instanceof Array){

File: src/main/routes/features/response/statementOfMeans/debts/debtsRouter.ts (hunk 5 above)
Comment (LeonardoPalmeiro) [reply src/main/routes/features/response/statementOfMeans/debts/debtsRouter.ts:13]: my bad. sorry.
I renamed

File: src/test/unit/routes/features/response/statementOfMeans/debts/debtsRouter.test.ts (hunk 1 above)
Comment (LeonardoPalmeiro) [reply src/test/unit/routes/features/response/statementOfMeans/debts/debtsRouter.test.ts:42]: done

File: src/test/unit/routes/features/response/statementOfMeans/debts/debtsRouter.test.ts (hunk 2 above)
Comment (LeonardoPalmeiro) [reply src/test/unit/routes/features/response/statementOfMeans/debts/debtsRouter.test.ts:28]: thanks I don't need jest.mock('../../../../../../../main/modules/draft-store');
done.

File: src/main/common/form/models/form.ts [hunk 6]
```diff
@@ -15,11 +15,26 @@ export class Form {
...
+  hasChildren(): boolean{
+    return this.hasErrors() ? this.errors
+      .some((error) => error.children?.length > 0) : false;
+  }
+
+  getNestedErrors(item : ValidationError, validators: FormValidationError[]): void {
+    item.children
+      .forEach(error => error.children
+        .forEach(childError => validators.push(new FormValidationError(childError, `${item.property}[${error.property}]`))));
+    //return validators;
```
Comment (TatsyMaslova) [inline src/main/common/form/models/form.ts:27]: perhaps we need to remove this if it is commented out?

Comment (LeonardoPalmeiro) [reply src/main/common/form/models/form.ts:27]: that's right!!
done

File: src/main/common/form/validators/formValidator.ts (hunk 4 above)
Comment (TatsyMaslova) [reply src/main/common/form/validators/formValidator.ts:8]: yes but there you are calling object.keys(form) again and this time you are getting the keys. on this particular line you are calling object.keys(form) but not doing anything with the keys

Comment (LeonardoPalmeiro) [reply src/main/common/form/validators/formValidator.ts:8]: thanks
done.

**Review by TatsyMaslova — APPROVED** [review]

**Pull request merged**